addons/binding/org.openhab.binding.vallox/src/main/java/org/openhab/binding/vallox/internal/serial/ValloxSerialInterface.java

```diff
@@ -0,0 +1,465 @@
+                    Telegram.byteToHex(telegram[4]), Telegram.byteToHex(telegram[5]));
+        }
+        serialWrite(telegram);
+    }
+
+    /**
+     * Send one telegram to the master unit.
+     * Example: send(Variable.FAN_SPEED.key, Telegram.convertBackFanSpeed((byte)4));
+     *
+     * @param variable one of the Variable keys in {@link Variable}}
+     * @param value the right binary coded value (see {@link Telegram} for some conversions)
+     * @throws IOException
+     */
+    public void send(byte variable, byte value) throws IOException {
+        send(variable, value, ValloxProtocol.ADDRESS_MASTER);
+    }
+
+    private void serialWrite(byte[] telegram) {
+        try {
+            if (outputStream != null) {
+                outputStream.write(telegram);
+                outputStream.flush();
+            }
+        } catch (IOException e) {
+            String msg = "Exception writing to vallox.";
+            logger.error(msg, e);
+            for (StatusChangeListener l : this.getStatusListener()) {
+                l.statusChanged(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, msg + " " + e.toString());
+            }
+        }
+    }
+
+    static byte calculateChecksum(byte[] pTelegram) {
+        int checksum = 0;
+        for (byte i = 0; i < pTelegram.length - 1; i++) {
+            checksum += pTelegram[i];
+        }
+        return (byte) (checksum % 256);
+    }
+
+    /**
+     * Start listening to telegrams on the serial interface. Stop if requested.
+     */
+    public void startListening() {
+        shutdownListening = false;
+        scheduler.submit(() -> {
+            long sleep = 0;
+            while (!shutdownListening && !Thread.interrupted()) {
+                try {
+                    Telegram telegram = receive();
+                    logger.trace("Received message: {}", telegram);
+                    telegram.updateStore(vallox, valueListener);
+                } catch (MalformedTelegramException e) {
+                    logger.warn("Issue receiving telegram. Discarding.", e);
+                } catch (InvalidRecepientException e) {
+                    logger.debug("Issue receiving telegram. Discarding.", e);
+                } catch (InsufficientDataException e) {
+                    // telegram too short, wait for more data, but ignore
+                    sleep = 200;
+                } catch (IOException e) {
+                    logger.error("Exception reading input stream.", e);
+                    // in case of low-level errors, retry, but not too quickly
+                    sleep = 2000;
+                } finally {
+                    if (sleep > 0) {
+                        try {
+                            Thread.sleep(sleep);
+                            sleep = 0; // reset sleep for the next telegram
+                        } catch (InterruptedException e1) {
+                            logger.error("Telegram listening thread interrupted.");
```

Comment:
Is this a bug situation that users should report as an issue if they see this or could it happen during a regular system shutdown?